### PR TITLE
Adds detailing functions for Slab reinforcement

### PR DESCRIPTION
**Pull request by ShivCharanSharma**
- adds detailing functions for L-Shape, U-Shape, Bent shape,
  Straight rebars for slab reinforcement
- Fixed default value of orientation in makeBentShapeRebar

### CODE REVIEW

File: SlabReinforcement/SlabReinforcement.py [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+    diameter,
+    amount_spacing_check,
+    amount_spacing_value,
+    rounding=2,
+    mesh_cover_along="Bottom",
+    structure=None,
+    facename=None,
+    bent_bar_length=50,
+    bent_bar_angle=135,
+):
```
Comment (amrit3701) [inline SlabReinforcement/SlabReinforcement.py:58]: you `typing` annotation for arguments. 

https://docs.python.org/3/library/typing.html

File: RebarData.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 import enum
 
 
-class RebarTypes(enum.Enum):
+class RebarTypes(str, enum.Enum):
```
Comment (amrit3701) [inline RebarData.py:32]: Nice. String Enum +1

**Review by amrit3701 — CHANGES_REQUESTED** [review]

File: SlabReinforcement/SlabReinforcement.py (hunk 1 above)
Comment (ShivCharanSharma) [reply SlabReinforcement/SlabReinforcement.py]: Added typing annotations, here is commit: https://github.com/amrit3701/FreeCAD-Reinforcement/pull/156/commits/5357f6de322e4c25b57e7b47660b1e125b330e89

File: RebarData.py (hunk 2 above)
Comment (ShivCharanSharma) [reply RebarData.py:32]: Thanks

File: SlabReinforcement/SlabReinforcement.py (hunk 1 above)
Comment (amrit3701) [reply SlabReinforcement/SlabReinforcement.py]: nice

**Review by amrit3701 — APPROVED** [review]

**Pull request merged**